packages/features/ee/cal-ai-phone/providers/retell-ai/RetellAIService.ts

```diff
@@ -0,0 +1,841 @@
+import { getStripeCustomerIdFromUserId } from "@calcom/app-store/stripepayment/lib/customer";
+import { getPhoneNumberMonthlyPriceId } from "@calcom/app-store/stripepayment/lib/utils";
+import stripe from "@calcom/features/ee/payments/server/stripe";
+import { checkRateLimitAndThrowError } from "@calcom/lib/checkRateLimitAndThrowError";
+import { WEBAPP_URL, IS_PRODUCTION } from "@calcom/lib/constants";
+import prisma from "@calcom/prisma";
+import { PhoneNumberSubscriptionStatus } from "@calcom/prisma/enums";
+
+import { TRPCError } from "@trpc/server";
+
+import type {
+  AIPhoneServiceUpdateModelParams,
+  AIPhoneServiceCreatePhoneNumberParams,
+  AIPhoneServiceImportPhoneNumberParamsExtended,
+} from "../../interfaces/ai-phone-service.interface";
+import { DEFAULT_BEGIN_MESSAGE, DEFAULT_PROMPT_VALUE } from "../../promptTemplates";
+import { RetellServiceMapper } from "./service-mappers";
+import type {
+  RetellLLM,
+  RetellCall,
+  RetellAgent,
+  RetellPhoneNumber,
+  RetellDynamicVariables,
+  AIConfigurationSetup,
+  AIConfigurationDeletion,
+  DeletionResult,
+  RetellAIRepository,
+  RetellLLMGeneralTools,
+  Language,
+} from "./types";
+import { getLlmId } from "./types";
+
+const MIN_CREDIT_REQUIRED_FOR_TEST_CALL = 5;
+
+export class RetellAIService {
+  constructor(private repository: RetellAIRepository) {}
+
+  async setupAIConfiguration(config: AIConfigurationSetup): Promise<{ llmId: string; agentId: string }> {
+    const generalTools = RetellServiceMapper.buildGeneralTools(config);
+
+    const llmRequest = RetellServiceMapper.mapToCreateLLMRequest(
+      {
+        ...config,
+        generalPrompt: config.generalPrompt || DEFAULT_PROMPT_VALUE,
+        beginMessage: config.beginMessage || DEFAULT_BEGIN_MESSAGE,
+      },
+      generalTools
+    );
+    const llm = await this.repository.createLLM(llmRequest);
+
+    const agentRequest = RetellServiceMapper.mapToCreateAgentRequest(llm.llm_id, config.eventTypeId);
+    const agent = await this.repository.createAgent(agentRequest);
+
+    return { llmId: llm.llm_id, agentId: agent.agent_id };
+  }
+
+  async importPhoneNumber(data: AIPhoneServiceImportPhoneNumberParamsExtended): Promise<RetellPhoneNumber> {
+    const { userId, agentId, teamId, ...rest } = data;
+    const { AgentRepository } = await import("@calcom/lib/server/repository/agent");
+
+    // Pre-check team permissions outside transaction
+    if (teamId) {
+      const canManage = await AgentRepository.canManageTeamResources({
+        userId,
+        teamId,
+      });
```

Comment:
I think provider shouldn't directly access the Application's repository. 

Better approach could be to return the data in an agreed upon format and then application then should access AgentRepository.


Or we could inject an agentRepository with an agreed upon format(providing only those APIs that provider needs maybe) that the provider can access and that agentRepository would then interact with the Application's AgentRepository which can do DB calls.

Reply:
Yeah i was thinking if i should inject it like retellRepository